pkg/data.go

```diff
@@ -128,14 +128,16 @@ func appendLog(tx *badger.Txn, identity *[64]byte, event UserEvent) error {
 	} else if err != nil {
 		return errorf(ErrDatabaseError, "%s", err)
 	} else {
-		data, err := item.Value()
+		err := item.Value(func(data []byte) error {
+			err = json.Unmarshal(data, currLog)
+			if err != nil {
+				return errorf(ErrDatabaseError, "invalid user log: %s", err)
```

Comment:
We don't need to wrap this error with `ErrDatabaseError` because the next error handler already does it.

Reply:
done